client.go

```diff
@@ -301,6 +304,19 @@ func NewClient(options ClientOptions) (*Client, error) {
 		options.MaxSpans = defaultMaxSpans
 	}
 
+	switch options.TracesInstrumenter {
+	case "":
+		options.TracesInstrumenter = "sentry"
+	case "sentry":
+		// noop
+	case "otel":
+		// sampling is performed by the OpenTelemetry SDK
+		options.TracesSampleRate = 1.0
+		options.TracesSampler = nil
+	default:
+		return nil, fmt.Errorf("invalid value for TracesInstrumenter (supported are 'sentry' and 'otel'): %q", options.TracesInstrumenter)
+	}
+
```

Comment:
I'm not sure if we should do this. I would rather have people define this explicitly in their options.

Reply:
you mean the `TracesSampleRate` and `TracesSampler`? Should we then error if the values are set to anything else? Doing neither seems a bit error-prone: `Instrumenter: "otel"` + `TracesSampleRate: 0` (default) would ignore otel sampling decisions and never send traces.

Reply:
Are you sure it does this? The span processor actually takes the Otel sampling decision into account, taking priority over the traces sample rate and traces sampler.

Reply:
I believe so. That might be a slightly different issue, though. One I had addressed in a [previous commit](https://github.com/getsentry/sentry-go/commit/9b18d57b992a06b384986d5e6faffef1871a293c#diff-27f9ebae19f3d3f9ca89ecaccdc8d8472ba81f86392efcb74e386532d26ca18a).
The current span processor considers only the sampled decision of the _parent_, so it only actually respects it if the span is part of an incoming trace. If it's a new trace, without a parent, the otel span might be sampled, but the sentry span remains at `SampleUndefined`, which then means we fall back to the client-wide sample rate.

[The current test](https://github.com/getsentry/sentry-go/blob/4c8f48a520c411e05569b35b0ade96310cdfb7af/otel/span_processor_test.go#L121) doesn't catch it because both the sentry client and the otel stack are initialized with "always trace", so they both arrive at the same decision, even though the processor isn't doing anything.

I just added a commit with the fixed test _and_ the fixed code. PTAL.

Reply:
friendly ping :innocent:

Reply:
I'll take a look later this week.

We made similar changes for node.js in https://github.com/getsentry/sentry-javascript/pull/9203, but I need to dive into this whole topic a bit deeper to give you actionable feedback.

Reply:
friendly ping :innocent: (I know these things get lower prio; just wanna avoid it getting forgotten)

Reply:
Yeah, sorry for that! I had to get some stuff shipped over on the PHP side of things, but the next few weeks will be again Go focus time for me.